x/leverage/keeper/msg_server.go

```diff
@@ -108,6 +108,51 @@ func (s msgServer) Collateralize(
 	return &types.MsgCollateralizeResponse{}, err
 }
 
+func (s msgServer) SupplyCollateral(
+	goCtx context.Context,
+	msg *types.MsgSupplyCollateral,
+) (*types.MsgSupplyCollateralResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	supplierAddr, err := sdk.AccAddressFromBech32(msg.Supplier)
+	if err != nil {
+		return nil, err
+	}
+	uToken, err := s.keeper.Supply(ctx, supplierAddr, msg.Asset)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
how about calling `s.Supply` and `s.Collaterize` directly here (by creating the appropriate message)?